yarn-project/stdlib/src/update-checker/update-checker.ts

```diff
@@ -0,0 +1,151 @@
+
+    return new UpdateChecker(
+      config.baseURL,
+      config.nodeVersion ?? getPackageVersion(),
+      config.rollupVersion ?? 'canonical',
+      rollupAddress,
+      config.fetch ?? fetch,
+      version => registryContract.getRollupAddress(version),
+      config.checkIntervalMs,
+    );
+  }
+
+  public async start() {
+    if (this.runningPromise.isRunning()) {
+      this.log.debug(`Can't start update checker again`);
+      return;
+    }
+
+    this.runningPromise.start();
+  }
+
+  public stop() {
+    if (!this.runningPromise.isRunning()) {
+      this.log.debug(`Can't stop update checker because it is not running`);
+      return;
+    }
+    return this.runningPromise.stop();
+  }
+
+  public trigger() {
+    return this.runningPromise.trigger();
+  }
+
+  private runChecks = async (): Promise<void> => {
+    await Promise.all([this.checkRollupVersion(), this.checkConfig()]);
+  };
+
+  private async checkRollupVersion(): Promise<void> {
+    try {
+      const canonicalRollup = await this.getRollupAddress(this.rollupVersion);
+      if (!canonicalRollup.equals(this.rollupAddressAtStart)) {
+        this.emit('newRollup', { currentRollup: this.rollupAddressAtStart, latestRollup: canonicalRollup });
+      }
+    } catch (err) {
+      this.log.warn(`Failed to check if there is a new rollup`, err);
+    }
+  }
+
+  private async checkConfig(): Promise<void> {
+    try {
+      const response = await this.fetch(
+        new URL('aztec-' + this.rollupAddressAtStart.toString() + '/index.json', this.configBasePath),
+      );
+      const body = await response.json();
+      if (!response.ok) {
+        if (response.status === 404) {
+          // no-op
+          return;
+        }
+        this.log.warn(`Unexpected HTTP response checking for updates`, {
+          status: response.status,
+          body: await response.text(),
+        });
+      }
+
+      const { version, config } = updateConfigSchema.parse(body);
+
+      if (version && version !== this.nodeVersion) {
+        this.emit('newVersion', { currentVersion: this.nodeVersion, latestVersion: version });
+      }
+
+      if (config && Object.keys(config).length > 0 && !isDeepStrictEqual(config, this.lastPatchedConfig)) {
+        this.lastPatchedConfig = config;
+        this.emit('updateConfig', config);
+      }
+    } catch (err) {
+      this.log.warn(`Failed to check if there is an update`, err);
+    }
+  }
+}
+
+/**
+ * Returns package version.
+ */
+export function getPackageVersion() {
+  try {
+    const releasePleaseManifestPath = resolve(
+      dirname(fileURLToPath(import.meta.url)),
+      '../../../../.release-please-manifest.json',
+    );
+    const version = JSON.parse(readFileSync(releasePleaseManifestPath).toString());
+    const buildMeta = process.env.BUILD_METADATA;
+    return buildMeta ? version['.'] + '-' + buildMeta : version['.'];
+  } catch (err) {
+    return '0.0.0';
```

Comment:
Same as above (not sure if the comment got deleted). If we fall into this case, the node will end up in a restart loop, since 0.0.0 will never be equal to the version on the index.json. We should return undefined here and skip the check if the local version is undefined.